text/0000-tcpconnection-hard-close.md

```diff
@@ -0,0 +1,52 @@
+- Feature Name: tcpconnection-hard-close
+- Start Date: 2019-03-03
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+Offer a way to immediately hard close a `TCPConnection`.
+
+# Motivation
+
+It is plausible that one may want to immediately hard close the `TCPConnection`
+from within the methods on `TCPConnectionNotify`. Currently however, it is not
+possible to hard close the connection as `TCPConnection.close()` only performs a
+hard close if the connection is muted. Muting the connection however is only
+available via the `TCPConnection.mute()` behavior that is run asynchronously.
+Hence one cannot call `conn.mute()` followed by `conn.close()` in order to hard
+close the connection.
+
+# Detailed design
+
+A new public method `fun ref hard_close()` would be created on `TCPConnection`.
+This method would invoke the existing private method `fun ref _hard_close()`
```

Comment:
Why not make the existing method public instead?

Reply:
Updated to make the existing method public.